### ISSUE TITLE
flash attention finite MASK_VALUE (-10000) lets masked keys dominate when valid logits fall below the sentinel

### ISSUE THREAD

**Issue by ohdearquant** [ohdearquant/lattice#361]
> _Issue filed by Claude (Anthropic agent) on behalf of @ohdearquant._

## Summary

`crates/inference/src/attention/flash.rs` masks padded keys with a **finite**
sentinel `const MASK_VALUE: f32 = -10_000.0` (flash.rs:14). In the online-softmax
loop (`tiled_attention_head` / `tiled_multi_head_attention`), a masked key's score
is *replaced* with `MASK_VALUE` and folded into `local_max`/`tile_sum`:

```rust
// flash.rs:374
let mut value = row[c] * scale;
if attention_mask[kv_start + c] == 0 {
    value = MASK_VALUE;
}
row[c] = value;
local_max = local_max.max(value);
```

If a **valid** (unmasked) score is below `-10000`, the masked sentinel becomes the
row max and the masked key receives nonzero — possibly dominant — probability.

## Concrete trigger

```text
seq_len = 2, head_dim = 1, scale = 1.0
q = [-20000.0, 0.0]
k = [1.0, 0.0]
v = [1.0, 100.0]
attention_mask = [1, 0]      # key 1 is masked
```

Query row 0: valid key score = `-20000`, masked key replaced with `-10000`.
Row max = `-10000` (the *masked* key). After exp, probabilities collapse to
`[0.0, 1.0]`.

- **Wrong:** output = `0.0*1.0 + 1.0*100.0 = 100.0` (the masked value wins).
- **Correct:** masked key probability must be exactly 0 → output = `1.0`.

## Severity / blast radius

Latent. Real models produce attention scores of O(10)–O(100), well above `-10000`,
so normal greedy/sampled inference is unaffected (this is why existing tests, which
use the same finite sentinel in their reference path, don't catch it). It bites only
on out-of-distribution / hostile / corrupt activations where a valid logit drops
below the sentinel. Filed for correctness-hardening, not as a release blocker.

## Suggested fix (needs care — NaN risk)

Track masked keys structurally in the online-softmax loop: skip masked columns when
computing `local_max`, `tile_sum`, and the weighted-value accumulation; **or** use
`f32::NEG_INFINITY` as the sentinel with explicit all-masked-row handling so
`-inf - -inf` in the `exp(value - new_max)` step does not produce NaN. The
all-masked-row case must be handled (define a zero/uniform output rather than NaN).

Found by an automated discovery pass over the attention module; verified by
direct reasoning against the source (no fix attempted here).
